backend/src/game-stat/game-stat.resolver.spec.ts

```diff
@@ -1,19 +1,720 @@
+      })
+      it('invalid number - infinity', async () => {
+        const data = {
+          loserId: 'aaaValidIdToTestuId1a',
+          winnerId: 'abaValidIdToTestuId1a',
+          type: EGameType.Classic,
+          scoreLoser: 1,
+          scoreWinner: 10,
+          timePlayed: Infinity
+        }
+        const metadata: ArgumentMetadata = {
+          type: 'body',
+          metatype: CreateGameStatInput,
+          data: ''
+        }
+        const res = {
+          message: ['timePlayed must be a number.'],
+          error: 'Bad Request',
+          statusCode: 400
+        }
+        const thrownError = await validationPipe
+          .transform(data, metadata)
+          .catch((error) => error)
+        expect(thrownError.getResponse()).toStrictEqual(res)
+      })
+      it('invalid number - Nan', async () => {
+        const data = {
+          loserId: 'aaaValidIdToTestuId1a',
+          winnerId: 'abaValidIdToTestuId1a',
+          type: EGameType.Classic,
+          scoreLoser: 1,
+          scoreWinner: 10,
+          timePlayed: NaN
+        }
+        const metadata: ArgumentMetadata = {
+          type: 'body',
+          metatype: CreateGameStatInput,
+          data: ''
+        }
+        const res = {
+          message: [
+            'timePlayed must not be less than 0.',
+            'timePlayed must be a number.'
+          ],
+          error: 'Bad Request',
+          statusCode: 400
+        }
+        const thrownError = await validationPipe
+          .transform(data, metadata)
+          .catch((error) => error)
+        expect(thrownError.getResponse()).toStrictEqual(res)
+      })
+      it('too low number', async () => {
+        const data = {
+          loserId: 'aaaValidIdToTestuId1a',
+          winnerId: 'abaValidIdToTestuId1a',
+          type: EGameType.Classic,
+          scoreLoser: 1,
+          scoreWinner: 10,
+          timePlayed: -1
+        }
+        const metadata: ArgumentMetadata = {
+          type: 'body',
+          metatype: CreateGameStatInput,
+          data: ''
+        }
+        const res = {
+          message: ['timePlayed must not be less than 0.'],
+          error: 'Bad Request',
+          statusCode: 400
+        }
+        const thrownError = await validationPipe
+          .transform(data, metadata)
+          .catch((error) => error)
+        expect(thrownError.getResponse()).toStrictEqual(res)
+      })
+    })
+    describe('type - EGameType tests', () => {
+      it('invalid type - Listambournian', async () => {
+        const data = {
+          loserId: 'aaaValidIdToTestuId1a',
+          winnerId: 'abaValidIdToTestuId1a',
+          type: 'Listambournian',
+          scoreLoser: 1,
+          scoreWinner: 10,
+          timePlayed: 10
+        }
+        const metadata: ArgumentMetadata = {
+          type: 'body',
+          metatype: CreateGameStatInput,
+          data: ''
+        }
+        const res = {
```

Comment:
Here you have to replace $constraint1 in dto's message by EGameType:
```ts
@IsEnum(EGameType, { message: '$property must be a valid EGameType.' })
```
And replace: 
```ts
['type must be a valid [object Object].']
```
by
```ts
['type must be a valid EGameStat.']
```
In the error message of the test